graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -98,18 +98,50 @@ class HeterogeneousLoaderConfig:
     """
 
     add_reverse_edges: bool
-    data_configs: Dict[str, Sequence[StructureConfig]]
+    data_configs: Mapping[str, Sequence[StructureConfig]]
     enable_assertions: bool
     graph_name: str
     input_prefix: str
     local_input_path: str
-    local_output_path: str
+    local_metadata_output_path: str
     num_output_files: int
     output_prefix: str
     precomputed_transformations: dict
 
 
-class DistHeterogeneousGraphLoader(HeterogeneousGraphLoader):
+@dataclass
+class ProcessedGraphRepresentation:
+    """JSON representations of a processed graph.
+
+    Parameters
+    ----------
+    processed_graph_metadata_dict : dict
+        A dictionary of metadata for the graph, in "chunked-graph"
+        format, with additional key "graph_info" that contains a more
+        verbose representation of th processed graph.
+
+        The dict also contains a "raw_id_mappings" key, which is a dict
+        of dicts, one for each node type. Each entry contains files information
+        about the raw-to-integer ID mapping for each node.
+
+        The returned value also contains an additional dict of dicts,
+        "graph_info" which contains additional information about the
+        graph in a more readable format.
+
+        For chunked graph format see
+        https://docs.dgl.ai/guide/distributed-preprocessing.html#specification
+    transformation_representations : dict
+        A dictionary containing the processed graph transformations.
+    timers : dict
+        A dictionary containing the timing information for different steps of processing.
+    """
+
+    processed_graph_metadata_dict: dict
```

Comment:
Previously, the only output is processed_graph_metadata_dict?
But now we have transformation_representations and timers?

Reply:
The function started with no return value, then with the processed_graph_metadata_dict, then added the timers, now we are adding a third dict.

I thought it's better to encapsulate everything in a class so 
1. We don't keep modifying the return type of the function
2. It's more clear what each item is, e.g. instead of return_val[1] we say  return_val.timers which clearly communicates what is inside the returned object.

Reply:
That makes sense